### PR TITLE
expose api ngx_http_upstream_check_upstream_down to check upstream

### PR DESCRIPTION
**Pull request by detailyang**
Hello, this PR is try to expose api for check upstream status on ngx_http_upstream_check_module for this issue #740

### CODE REVIEW

File: src/http/ngx_http_upstream_check_module.c [hunk 1]
```diff
@@ -1321,6 +1321,29 @@ ngx_http_upstream_check_peer_down(ngx_uint_t index)
...
+    ngx_uint_t i;
+    ngx_http_upstream_check_peer_t *peers;
+
+    if (check_peers_ctx == NULL) {
+        return 0;
+    }
+
+    peers = check_peers_ctx->peers.elts;
+    for (i = 0; i < check_peers_ctx->peers.nelts; i++) {
+        if (ngx_strncmp(peers[i].upstream_name->data, upstream->data, upstream->len) == 0) {
```
Comment (chobits) [inline src/http/ngx_http_upstream_check_module.c]: also need to check `peers[i].upstream_name->len == upstream->len`

Comment (detailyang) [inline src/http/ngx_http_upstream_check_module.c]: fixed :beers:

File: src/http/ngx_http_upstream_check_module.c [hunk 2]
```diff
@@ -1321,6 +1321,30 @@ ngx_http_upstream_check_peer_down(ngx_uint_t index)
...
+    ngx_http_upstream_check_peer_t *peers;
+
+    if (check_peers_ctx == NULL) {
+        return 0;
+    }
+
+    peers = check_peers_ctx->peers.elts;
+    for (i = 0; i < check_peers_ctx->peers.nelts; i++) {
+        if (peers[i].upstream_name->len == upstream->len
+            && ngx_strncmp(peers[i].upstream_name->data, upstream->data, upstream->len) == 0) {
```
Comment (chobits) [inline src/http/ngx_http_upstream_check_module.c]: with nginx coding style, it shoud be as following:

```
 if (...
     && ...)
 {                    // `{' should starts in a new line aligns with `if'.
      ...
 }
```

Comment (detailyang) [inline src/http/ngx_http_upstream_check_module.c]: gotcha

**Comment by chobits** [comment]
hi @detailyang , you can add a function declaration into `#if (NGX_HTTP_UPSTREAM_CHECK) macro` in `src/http/ngx_http_upstream.h` .

**Comment by detailyang** [comment]
@chobits sorry for about function declaration, now it's okay :grin:

**Comment by chobits** [comment]
useful API, merged, thanks.

**Pull request merged**